napps/kytos/of_core/main.py

```diff
@@ -145,53 +147,101 @@ def handle_echo_request(self, event):
         """
         log.debug("Echo Request message read")
 
-        echo_request = event.content['message']
+        echo_request = event.message
         echo_reply = EchoReply(xid=echo_request.header.xid)
         event_out = KytosEvent(name=('kytos/of_core.messages.out.'
                                      'ofpt_echo_reply'),
                                content={'message': echo_reply,
                                         'destination': event.source})
         self.controller.buffers.msg_out.put(event_out)
 
+    @listen_to('kytos/core.connection.new')
+    def handle_core_new_connection(self, event):
+        """Method called when a new switch is connected.
+
+        This method will send a hello world message when a new switch is
+        connected.
+        """
+        self.say_hello(event.source)
```

Comment:
@erickvermot , ok, I agree with your argument.

Just one question [really, I don't remember this point from the spec].
If the `Hello` messages are sent independently one from each other, when one knows that the handshake process should proceed? I mean, how both the switch and the controller knows that they should stop sending `Hello` messages to each other?
I talked about the `xid` because it seems to me that this would be a "marker" on the handshake negotiation.
For example:

> **Switch**:
>     - Hi Controller, I want to talk OF1.3 with you. *[Hello with xid=123]*
> **Controller**:
>     - Hi Switch, you have sent me a Hello message saying that you want to talk OF1.3.
>     - I'm answering you that I only talk OF1.2. *[Hello with xid=123]*
> **Switch**:
>     - Hi Controller, thanks for the answer *[Hello with xid=123]*. Ok then, let's talk OF1.2

(Ok, this may have been a little bit ridiculous (rs) but I hope I have made my doubt clear)